File: src/main/java/hzt/aoc/day08/Part1HandheldHalting.java
package hzt.aoc.day08;

import java.util.List;

public class Part1HandheldHalting extends Day08Challenge {

    public Part1HandheldHalting() {
        super("Handheld Halting part 1",
                "Immediately before any instruction is executed a second time, " +
                        "what value is in the accumulator?");
    }


    @Override
    protected int solveByInstructions(List<Instruction> instructions) {
        return testInstructions(instructions).getGlobal();
    }

    public String getMessage(int global) {
        return String.format("The value of the global variable before second execution: %d%n", global);
    }

}

File: src/main/kotlin/hzt/aoc/day08/Part1HandheldHalting.kt
package hzt.aoc.day08

class Part1HandheldHalting : Day08Challenge(
    "part 1",
    "Immediately before any instruction is executed a second time, " +
            "what value is in the accumulator"
) {
    override fun solveByInstructions(instructions: List<Instruction>): Int {
        return testInstructions(instructions).global
    }

    override fun getMessage(result: String): String {
        return String.format("The value of the global variable before second execution: %s%n", result)
    }
}
